src/main/java/com/hththn/dev/department_manager/service/UserDetailsCustom.java

```diff
@@ -0,0 +1,35 @@
+package com.hththn.dev.department_manager.service;
+
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+
+
+@Component("userDetailsService")
+public class UserDetailsCustom implements UserDetailsService {
```

Comment:
Không thấy sử dụng tới Service này trong toàn bộ source code

Reply:
đây là triển khai của UserDetailsService của SpringSecurity. nhiệm vụ là tìm và trả về đối tượng UserDetails từ csdl.